src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/apiUsage/PrintfScanfArgumentsInspector.java

```diff
@@ -55,10 +55,13 @@ public String getDisplayName() {
         functions.put("fscanf",  1);
     }
 
-    final static private Pattern regexPlaceHolders;
+    final static private Pattern printfRegexPlaceHolders;
+    final static private Pattern scanfRegexPlaceHolders;
+
     static {
         // raw regex: %((\d+)\$)?[+-]?(?:[ 0]|\\?'.)?-?\d*(?:\.\d+)?[\[sducoxXbgGeEfF]
-        regexPlaceHolders = Pattern.compile("%((\\d+)\\$)?[+-]?(?:[ 0]|\\\\?'.)?-?\\d*(?:\\.\\d+)?[\\[sducoxXbgGeEfF]");
+        printfRegexPlaceHolders = Pattern.compile("%((\\d+)\\$)?[+-]?(?:[ 0]|\\\\?'.)?-?\\d*(?:\\.\\d+)?[\\[sducoxXbgGeEfF]");
+        scanfRegexPlaceHolders  = Pattern.compile("%\\*?((\\d+)\\$)?[+-]?(?:[ 0]|\\\\?'.)?-?\\d*(?:\\.\\d+)?[\\[sducoxXbgGeEfF]");
```

Comment:
@Powerhamster : I'd like to keep one regex only, can you refactor to turn '(\\d+)' into '(\\d+|\\*)'? That might be not 100% correct, but keeps inspection complexity on reasonable level.